File: src/page/basis/ComboList.java
package page.basis;

import java.awt.Component;
import java.util.List;

import javax.swing.DefaultListCellRenderer;
import javax.swing.JLabel;
import javax.swing.JList;

import util.Interpret;
import util.basis.Combo;

class ComboList extends JList<Combo> {

	private static final long serialVersionUID = 1L;

	protected ComboList() {
		setCellRenderer(new DefaultListCellRenderer() {

			private static final long serialVersionUID = 1L;

			@Override
			public Component getListCellRendererComponent(JList<?> l, Object o, int ind, boolean s, boolean f) {
				Combo c = (Combo) o;
				JLabel jl = (JLabel) super.getListCellRendererComponent(l, o, ind, s, f);
				jl.setText(Interpret.comboInfo(c));
				return jl;
			}

		});
	}

	protected void setList(List<Combo> lf) {
		setListData(lf.toArray(new Combo[0]));
	}

}

File: src/page/basis/ComboList.kt
package page.basis

import common.util.unit.Combo
import utilpc.Interpret
import java.awt.Component
import javax.swing.DefaultListCellRenderer
import javax.swing.JLabel
import javax.swing.JList

internal class ComboList : JList<Combo?>() {
    protected fun setList(lf: List<Combo?>) {
        setListData(lf.toTypedArray())
    }

    companion object {
        private const val serialVersionUID = 1L
    }

    init {
        setCellRenderer(object : DefaultListCellRenderer() {
            private static
            val serialVersionUID = 1L
            override fun getListCellRendererComponent(l: JList<*>?, o: Any?, ind: Int, s: Boolean, f: Boolean): Component {
                val c: Combo? = o as Combo?
                val jl = super.getListCellRendererComponent(l, o, ind, s, f) as JLabel
                jl.text = Interpret.comboInfo(c)
                return jl
            }
        })
    }
}
